backend/src/main/java/io/debezium/configserver/rest/client/jolokia/JolokiaClient.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.configserver.rest.client.jolokia;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import javax.management.MalformedObjectNameException;
+
+import io.debezium.configserver.model.JolokiaURIsList;
+import io.smallrye.config.SmallRyeConfig;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jolokia.client.BasicAuthenticator;
+import org.jolokia.client.J4pClient;
+import org.jolokia.client.exception.J4pException;
+import org.jolokia.client.request.J4pReadRequest;
+import org.jolokia.client.request.J4pReadResponse;
+
+@ApplicationScoped
+public class JolokiaClient {
+    public static final int DEFAULT_JOLOKIA_PORT = 8778;
+    public static final String PROPERTY_JOLOKIA_CLIENT_URI = "jolokia.client.uri";
+    private static JolokiaURIsList jolokiaURIsList;
+    @ConfigProperty(name = PROPERTY_JOLOKIA_CLIENT_URI)
+    private Optional<String> configJolokiaUrl;
+    @Inject
+    JolokiaAttributes jolokiaAttributes;
+    public static JolokiaURIsList getJolokiaBaseUris() {
+        if (jolokiaURIsList == null) {
+            List<String> jolokiaBaseUris = ((SmallRyeConfig) ConfigProvider.getConfig())
+                    .getValues(PROPERTY_JOLOKIA_CLIENT_URI, String.class, ArrayList::new);
```

Comment:
This way it is hard to relate a Kafka Connect cluster to a Jolokia URI. The Kafka Connect cluster config should accept two URIs: the KC URI and a optional second Jolokia URI. When this one is not set, default/fallback to the Kafka Connect URI and rewrite the HTTP port to the default Jolokia port and the path to the default path.

This is another reason why I think doing it like this (and doing it now) is over-engineered and takes more and more time to finish this task. And we already know the code is about to be removed again soon and replaced with something new.